Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -24,13 +23,99 @@
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
+#include "vtkExtentTranslator.h"
+
+
+#define USE_SMP_DEFAULT_2D_BLOCK
+
+// the default 2d block is used to compare with tbb's own 2d splitting option
+#ifdef USE_SMP_DEFAULT_2D_BLOCK
+  #include <tbb/parallel_for.h>
+  #include <tbb/blocked_range2d.h>
+#else
+  #include "vtkSMPTools.h"   
+#endif
+
+struct vtkImageThreadStruct
+{
+vtkThreadedImageAlgorithm *Filter;
+vtkInformation *Request;
+vtkInformationVector **InputsInfo;
+vtkInformationVector *OutputsInfo;
+vtkImageData   ***Inputs;
+vtkImageData   **Outputs;
+};
 
 
+#ifdef USE_SMP_DEFAULT_2D_BLOCK
```

Comment:
Don't indent after an #ifdef.  The #ifdef is for the preprocessor and therefore not part of the C++ grammar, so it should have no impact on the styling or formatting of the surrounding C++ code.